src/managedwriter/stream_connection.ts

```diff
@@ -87,6 +87,16 @@ export class StreamConnection extends EventEmitter {
     this._connection.on('error', this.handleError);
     this._connection.on('close', () => {
       this.trace('connection closed');
+      this.reconnect();
```

Comment:
Reconnect only when the error is retryable?

Reply:
now we only reconnect when there are pending writes, which will cover the case where the error is retryable.